### DISCUSSION TITLE
Rewrites with dynamic routes

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/next.js#13170] [Help]
I'm in need of outputting XML on one of my Next.js pages. As far as I understand it I can only do this from the `api` folder, but please correct me if I'm wrong.

Is there any way to rewrite my `/api/feed/[category].js` dynamic route so that it's found at `/feed/[category]`?

I tried:
```
module.exports ={
    experimental: {
      modern: true,
      async rewrites() {
        return [
          { source: "/feed", destination: "/api/feed" },
        ];
      },
      catchAllRouting: true
    }
  }
```

But any dynamic routing was lost unfortunately.

**Comment by ijjk** [comment]
Hi, to match segments with the rewrites feature and point to a dynamic page you need to specify them like this:

```js
module.exports = {
  experimental: {
    async rewrites() {
      return [
        {
          source: '/feed/:category',
          destination: '/api/feed/:category'
        }
      ]
    }
  }
}
```

which maps `/feed/hello` -> `/api/feed/[category].js`

**Reply by gtothesquare** [reply]
Hi! can one do this 
```
async rewrites() {
      return [
        {
          source: '/:locale/feed/:category',
          destination: '/feed/:category'
        }
      ]
    }
````
And get the `:locale` param somewhere ?

Basically is two fold, have a dynamic path rewrite at start and then get that param somewhere in the page that it lands.

**Reply by ijjk** [reply]
The `:locale` path param from the rewrite and the `category` from the dynamic route param should both be made available under the `query` object. This RFC also might of interest https://github.com/vercel/next.js/discussions/17078
